sway/desktop/output.c

```diff
@@ -38,19 +40,33 @@ static void output_frame_view(swayc_t *view, void *data) {
 	//		return;
 	//}
 
+	// if the shell specifies window geometry, make the top left corner of the
+	// window in the top left corner of the container to avoid arbitrarily
+	// sized gaps based on the attached buffer size
+	int window_offset_x = 0;
+	int window_offset_y = 0;
+
+	if (view->sway_view->type == SWAY_XDG_SHELL_V6_VIEW) {
+		window_offset_x = view->sway_view->wlr_xdg_surface_v6->geometry->x;
```

Comment:
Multiply this by the output scale to get output-local coordinates (`renmder{width,height}` are in output-local coordinates).

Reply:
I'm going to defer fixing this until my next effort, which will be to render popups and subsurfaces and have them work with inputs.